packages/aws-client/src/S3.ts

```diff
@@ -189,53 +224,27 @@ exports.downloadS3File = (s3Obj, filepath) => {
   });
 };
 
-
-/**
-* Get an object header from S3
-*
-* @param {string} Bucket - name of bucket
-* @param {string} Key - key for object (filepath + filename)
-* @param {Object} retryOptions - options to control retry behavior when an
-*   object does not exist. See https://github.com/tim-kos/node-retry#retryoperationoptions
-*   By default, retries will not be performed
-* @returns {Promise} - returns response from `S3.headObject` as a promise
-**/
-exports.headObject = improveStackTrace(
-  (Bucket, Key, retryOptions = { retries: 0 }) =>
-    pRetry(
-      async () => {
-        try {
-          return await awsServices.s3().headObject({ Bucket, Key }).promise();
-        } catch (err) {
-          if (err.code === 'NotFound') throw err;
-          throw new pRetry.AbortError(err);
-        }
-      },
-      { maxTimeout: 10000, ...retryOptions }
-    )
-);
-
 /**
  * Get the size of an S3Object, in bytes
  *
  * @param {string} bucket - S3 bucket
  * @param {string} key - S3 key
- * @returns {Promise<integer>} - object size, in bytes
+ * @returns {Promise<integer>} object size, in bytes
  */
-exports.getObjectSize = (bucket, key) =>
-  exports.headObject(bucket, key, { retries: 3 })
+export const getObjectSize = (bucket: string, key: string) =>
+  headObject(bucket, key, { retries: 3 })
     .then((response) => response.ContentLength);
 
 /**
 * Get object Tagging from S3
 *
 * @param {string} bucket - name of bucket
 * @param {string} key - key for object (filepath + filename)
-* @returns {Promise} - returns response from `S3.getObjectTagging` as a promise
+* @returns {Promise} returns response from `S3.getObjectTagging` as a promise
```

Comment:
Does AWS provide a class for the response type inside the promise?

Reply:
Updated